sklearn/feature_extraction/_hash.py

```diff
@@ -71,6 +71,12 @@ class FeatureHasher(TransformerMixin, BaseEstimator):
     DictVectorizer : Vectorizes string-valued features using a hash table.
     sklearn.preprocessing.OneHotEncoder : Handles nominal/categorical features.
 
+    Notes
+    -----
+    This estimator is stateless and does not need to be fitted. However, we
+    advice to class :meth:`fit_transform` instead of only :meth:`transform` to
+    benefit from the parameters validation that only happens in :meth:`fit`.
```

Comment:
```suggestion
    recommend to call :meth:`fit_transform` instead of :meth:`transform`, as
    parameter validation is only performed in :meth:`fit`.
```

WDYT?

I wonder if we can find a good word to make it clear that we are talking about the hyper-parameters (aka those passed to the constructor), not other parameters. Basically, I wonder if the average user would know which parameters this comment refers to without some more hints. But I also don't think we should add a whole new sentence to explain it. So yeah. Maybe someone has an idea.  Also find without it though.

Reply:
(also applies to the other instances of this doc string)

Reply:
The docstring of `__init__` has a "Parameters" section. It seems to provide enough context on what we refer to as parameters.

If we would have change the docstring of another class method, then I think that we should have been extremely explicit.